### DISCUSSION TITLE
If i am going to start watching this course, Should i need to Learn Python Before Starting this Course?

### DISCUSSION THREAD

**Discussion by Bharath-Anbazhagan** [Cyfrin/foundry-full-course-cu#1141] [Q&A] [answered]
I don't know any coding language. But i am interested to learn. But also more interested in learning Blockchain Dev. So, I need your help family 😇 

Let me what should be my roadmap for it

Thank You!

**Comment by Niferu** [comment, marked as answer]
Hi @Bharath-Anbazhagan 

If you want to be pure solidity developer (backend only) this course will be sufficient for you as there are no other languages than solidity involved. If you want to be full-stack developer then you might be interested of taking this version of the course [link](https://github.com/smartcontractkit/full-blockchain-solidity-course-js). In this one you will learn java script and/or type script additionally to solidity :)

**Reply by Bharath-Anbazhagan** [reply]
Hi @Nlferu,

Thank you so much for the guidance! I'm excited about delving into Solidity development. I think I'll start with the pure Solidity course to get a solid foundation. Once I'm comfortable, I might explore the full-stack development course to broaden my skills with JavaScript and/or TypeScript. Your input is invaluable, and I appreciate your help in shaping my learning journey.

Best,
@Bharath-Anbazhagan

**Reply by Niferu** [reply]
@Bharath-Anbazhagan 

I'm glad I could help. Additional information for you, I personally think pure solidity course is a bit harder than JS/TS version of it as here you will be writing everything in solidity, so interactions, deploy scripts and even config file or tests will be written in solidity, which is pretty challenging to be honest. In JS/TS version you would be writing all scripts/utils/tests in JS/TS languages, which is much clearer for beginners, so if you dont mind you can just start full-stack version and you can simply skip front-end parts :)

But it is just a tip, so you can chose best for you as starting with foundry course if you have 0 knowledge might be very hard. I have finished 3 versions of this course (Python + Solidity, JavaScript/TypeScript + Solidity, Foundry -> pure Solidity) and foundry one is hardest in my opinion, but it is most up to date also, so pick your best and good luck mate! :)

**Reply by Bharath-Anbazhagan** [reply]
Hi @Nlferu,

Thank you for the detailed insight! Your experience with the different versions of the course is really helpful. I'm actually more interested in learning Python, so the Python + Solidity version caught my eye. Do you think it's a good idea to start with Python, or would you recommend diving into JavaScript/TypeScript first and then picking up Python later?

Also, thanks for the tip about the challenges with the pure Solidity course. I'm considering the full-stack version but was wondering about the Python option.

Additionally, I'm curious about whether it's advisable to finish learning Python before diving into Solidity, or if it's feasible to learn both simultaneously. Your insights from completing the courses would be really valuable in helping me plan my learning journey.

Thanks again for your guidance!
@Bharath-Anbazhagan

**Reply by Niferu** [reply]
Hi @Bharath-Anbazhagan 

I think best option is to dive into JS/TS + Solidity as Python version of this course is HEAVILY outdated and you will keep facing a lot of issues with no help from community. If you do not know any programming language that is not the issue at all as Patrick is explaining things pretty good and you can learn all languages involved in course simultaneously (you will see that almost all programming languages syntax are very similar). I would recommend you diving into JS/TS + Solidity course and keep asking about everything that is not so clear for you to get most understanding how everything works :). Once you finish JS/TS version you should straight forward jump into solidity one and do it bit quicker as a lot of things will be just the same, but it will help you understand all much better. Once you are bored of learning or feeling comfortable with all you have learned so far, definitely you want to do some projects on your own as this will reward you most :) Solidity version of this course has also challenge for each lesson and those are pretty hard to solve but giving you tons of fun and knowledge while cracking them. Good luck mate! :)

**Reply by Niferu** [reply]
Ahh you also want to use Chat GPT as it can help you a lot fixing some issues :)

**Reply by Bharath-Anbazhagan** [reply]
Hi @Nlferu,

Thank you so much for your detailed advice! I really appreciate your insights, and it's clear that diving into JS/TS + Solidity is the way to go. Your guidance has been invaluable.

Since I've decided to start with JS/TS, could you please recommend the best free resources for learning JavaScript/TypeScript in your opinion? I want to make sure I have a solid foundation before moving on to Solidity.

Additionally, I'm curious if you know of any free certification courses for learning JavaScript/TypeScript, especially those focused on blockchain development. Your recommendations would be immensely helpful as I'm keen on acquiring a certification as I progress in my learning journey.

Thanks again!
@Bharath-Anbazhagan

**Reply by Niferu** [reply]
Hi @Bharath-Anbazhagan 

Unfortunately I do not know any courses, which can give you certs. i was learning everything from Patrick as I had no JS/TS exp too, but this guy here is amazing [LINK](https://www.youtube.com/watch?v=Yl7ULiRhhdU&ab_channel=ByteGrad) he is explaining everything with extra details but he is also focused for front-end. Personally I would just start with Patrick course, it will be harder without knowing basics yes but it is possible as I have passed all of his courses without knowing any programming mainstream language :) But maybe someone else here can help you by advising which course to take to get certs and most of it for pure JS/TS, just create new discussion and I'm sure someone will give you some good materials :)

**Reply by Bharath-Anbazhagan** [reply]
Hi @Nlferu,

Thank you so much for your support and the link you shared. I'll definitely check out the recommended YouTube videos for HTML, CSS, and JS.

I'm pleased to share that my friend, who is currently in college, has also suggested three YouTube videos that teach these languages in my native language, making it much easier for me to understand.

I've taken note of the blockchain developer roadmap, and it aligns perfectly with my plan to start with HTML, CSS, and JS. Your guidance has been instrumental in helping me feel confident about the path I'm on.

If you think it would be beneficial, I'd be happy to share my roadmap for feedback. Let me know if that would be helpful.

Thanks once again for your assistance! 💖
@Bharath-Anbazhagan

**Reply by Niferu** [reply]
@Bharath-Anbazhagan 

Sure thing please share your roadmap I will happily express my opinion on it :) And I'm glad i could help so kind person like you are, cheers!

**Comment by Bharath-Anbazhagan** [comment]
![RoadMap 1-7](https://github.com/Cyfrin/foundry-full-course-f23/assets/149758658/697d23b9-a8cb-47ca-a19c-744f71125ed9)
![RoadMap 8-14](https://github.com/Cyfrin/foundry-full-course-f23/assets/149758658/48b0d2f6-424a-490c-853b-165908d61f54)

**Reply by Niferu** [reply]
@Bharath-Anbazhagan 

You just must know that HTML/CSS and React/NextJS are pure front-end techs, but mostly it looks good :) you will also get everything in great order from JS/TS + Solidity course

**Reply by Bharath-Anbazhagan** [reply]
Hi @Nlferu,

Thanks for the heads up! I appreciate your clarification on HTML/CSS and React/NextJS being primarily front-end technologies. It's good to know that the JS/TS + Solidity course will cover everything in a comprehensive order.

I'm looking forward to diving into the course and gaining a deeper understanding of these technologies. Your insights have been incredibly helpful, and I'm grateful for your guidance.

If I have any more questions along the way, would it be okay to reach out to you?

Thanks again! 💖
@Bharath-Anbazhagan

**Reply by Niferu** [reply]
@Bharath-Anbazhagan 

Sure thing :)
